gtsfm/loader/hilti_loader.py

```diff
@@ -90,22 +93,29 @@ def __init__(
         if self._max_length is not None:
             self.num_rig_poses = min(self.num_rig_poses, self._max_length)
 
+        # Read GT poses from odom file:
+        self._fastlio_poses = self.__read_fastlio_poses()
+
         # Read the constraints from the lidar/constraints file
         all_constraints: Dict[Tuple[int, int], Constraint] = self.__load_constraints()
+        logger.info("[hilti_loader] Number of filtered constraints: %d", len(all_constraints))
         filtered_constraints: Dict[Tuple[int, int], Constraint] = self._filter_outlier_constraints(all_constraints)
         self._constraints: Dict[Tuple[int, int], Constraint] = self._update_stationary_constraints(filtered_constraints)
 
-        logger.info("Number of constraints: %d", len(self._constraints))
+        logger.info("[hilti_loader] Number of constraints: %d", len(self._constraints))
 
         # Read the poses for the IMU for rig indices from g2o file.
-        self._w_T_imu: Dict[int, Pose3] = self.__read_lidar_pose_priors()
+        self._w_T_imu: Dict[int, Pose3] = dict(
+            enumerate(self._fastlio_poses)
+        )  # Quick hack! self.__read_lidar_pose_priors()
 
-        logger.info("Loading %d timestamps", self.num_rig_poses)
-        logger.info("Lidar camera available for %d timestamps", len(self._w_T_imu))
+        logger.info("[hilti_loader] Loading %d timestamps", self.num_rig_poses)
+        logger.info("[hilti_loader] Lidar camera available for %d timestamps", len(self._w_T_imu))
 
     def __load_constraints(self) -> Dict[Tuple[int, int], Constraint]:
         constraints_path = self._base_folder / LIDAR_CONSTRAINTS_RELATIVE_PATH
-        constraints = Constraint.read(str(constraints_path))
+        all_constraints = Constraint.read(str(constraints_path))
+        constraints = PoseSlam.filter_constraints(all_constraints, self._fastlio_poses)
```

Comment:
can we do this where the other filtering code is called, just so it all sits together?

Reply:
OK